Rename "alternate root" terminology in projects and build configurations

davidquesada opened the pull request:
#### ce0fb9eaa887d4273a05931ccdac4992c9b8fb07
<pre>
Rename &quot;alternate root&quot; terminology in projects and build configurations
<a href="https://bugs.webkit.org/show_bug.cgi?id=288218">https://bugs.webkit.org/show_bug.cgi?id=288218</a>

Reviewed by Elliott Williams.

Deobfuscate &quot;alternate root&quot; and refer to the OS cryptex by name directly, since the
install layout of WebKit into this path is no longer a secret. Make the following renames:

    create-symlink-to-altroot.sh =&gt; create-symlink-to-cryptex.sh
    ALTERNATE_ROOT_PATH =&gt; CRYPTEX_SYMLINK_PATH
    OUTPUT_ALTERNATE_ROOT_DIR =&gt; OUTPUT_CRYPTEX_SYMLINK_DIR
    OUTPUT_ALTERNATE_ROOT_PATH =&gt; OUTPUT_CRYPTEX_SYMLINK_PATH
    &quot;Create Symlink to Alt Root Path&quot; script phase =&gt; &quot;Create Symlink to Cryptex Path&quot;

* Configurations/WebKitProjectPaths.xcconfig:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/JavaScriptCore/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/ThirdParty/libwebrtc/Scripts/create-symlink-to-cryptex.sh: Renamed from Source/ThirdParty/libwebrtc/Scripts/create-symlink-to-altroot.sh.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/WebCore/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/WebGPU/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebInspectorUI/Configurations/WebInspectorUIFramework.xcconfig:
* Source/WebInspectorUI/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/WebInspectorUI/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/WebInspectorUI/WebInspectorUI.xcodeproj/project.pbxproj:
* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Configurations/_WebKit_SwiftUI.xcconfig:
* Source/WebKit/Scripts/copy-launchd-plist-and-create-symlink.sh:
* Source/WebKit/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/WebKit/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/scripts/create-symlink-to-cryptex.sh: Renamed from Source/WebKitLegacy/scripts/create-symlink-to-altroot.sh.

Canonical link: <a href="https://commits.webkit.org/291920@main">https://commits.webkit.org/291920@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7a73ef10a3f99f1039464c70d3f024d2b3f1d4ac

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/93553 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13123 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/2857 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/98556 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44079 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/95603 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/13412 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/21569 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71472 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28848 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/96555 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10036 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84604 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51807 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/93048 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/9718 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/2227 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/43393 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/86263 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79993 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2285 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/100588 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/92219 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/20605 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15067 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80486 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/20857 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/80538 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79820 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20041 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24357 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/1703 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/13753 "Built successfully") | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15146 "") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/20589 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/25767 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/114869 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/20276 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/33088 "Found 9 new JSC stress test failures: microbenchmarks/memcpy-wasm-medium.js.bytecode-cache, microbenchmarks/memcpy-wasm-medium.js.mini-mode, stress/string-value-of-error.js.bytecode-cache, wasm.yaml/wasm/function-tests/memcpy-wasm-loop.js.default-wasm, wasm.yaml/wasm/stress/b3-signed-extend-16-to-64.js.wasm-eager-jettison, wasm.yaml/wasm/stress/ipint-bbq-osr-with-try5.js.wasm-slow-memory, wasm.yaml/wasm/stress/live-funcref-across-loop-tier-up.js.default-wasm, wasm.yaml/wasm/stress/simple-inline-exception-inlinee-catch-with-tag-arg.js.wasm-collect-continuously, wasm.yaml/wasm/stress/simple-inline-exception-inlinee-catch-with-tag-arg.js.wasm-eager (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/23736 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22017 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/JavaScriptCore/Scripts/create-symlink-to-cryptex.sh:
Unfortunately the renaming changes so many lines of the script that git doesn't track it as a rename. I'm assuming you didn't change anything meaningful in here?

davidquesada replied:
Right. With the exception of a few manually edited comments [0], all the changes here are scripted, mechanical renames. No refactoring or behavioral changes intended/expected here.

[0]
https://github.com/WebKit/WebKit/pull/41080/files#diff-1c5e8bbd62cd85e9da7967131a827393abb37e5844ccebb84a164b3ab6536414R50
https://github.com/WebKit/WebKit/pull/41080/files#diff-491f244e14e949063b4e952ced9c9f4fed2f18aedc80727fe18ccaa2cbefb108R18

emw-apple reviewed (approved):
rs=me

davidquesada committed:
Rename "alternate root" terminology in projects and build configurations
https://bugs.webkit.org/show_bug.cgi?id=288218

Reviewed by Elliott Williams.

Deobfuscate "alternate root" and refer to the OS cryptex by name directly, since the
install layout of WebKit into this path is no longer a secret. Make the following renames:

    create-symlink-to-altroot.sh => create-symlink-to-cryptex.sh
    ALTERNATE_ROOT_PATH => CRYPTEX_SYMLINK_PATH
    OUTPUT_ALTERNATE_ROOT_DIR => OUTPUT_CRYPTEX_SYMLINK_DIR
    OUTPUT_ALTERNATE_ROOT_PATH => OUTPUT_CRYPTEX_SYMLINK_PATH
    "Create Symlink to Alt Root Path" script phase => "Create Symlink to Cryptex Path"

* Configurations/WebKitProjectPaths.xcconfig:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/JavaScriptCore/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/ThirdParty/libwebrtc/Scripts/create-symlink-to-cryptex.sh: Renamed from Source/ThirdParty/libwebrtc/Scripts/create-symlink-to-altroot.sh.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/WebCore/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/WebGPU/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebInspectorUI/Configurations/WebInspectorUIFramework.xcconfig:
* Source/WebInspectorUI/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/WebInspectorUI/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/WebInspectorUI/WebInspectorUI.xcodeproj/project.pbxproj:
* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Configurations/_WebKit_SwiftUI.xcconfig:
* Source/WebKit/Scripts/copy-launchd-plist-and-create-symlink.sh:
* Source/WebKit/Scripts/create-symlink-to-altroot.sh: Removed.
* Source/WebKit/Scripts/create-symlink-to-cryptex.sh: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Source/WebKitLegacy/scripts/create-symlink-to-cryptex.sh: Renamed from Source/WebKitLegacy/scripts/create-symlink-to-altroot.sh.

Canonical link: https://commits.webkit.org/291920@main
